src/bloqade/analysis/common/assignment_scan.py

```diff
@@ -12,7 +13,12 @@ def visit_waveform_Record(self, node: waveform.Record):
         self.visit(node.waveform)
         duration = node.waveform.duration(**self.assignments)
         var = node.var
-        value = node.waveform.eval_decimal(duration, **self.assignments)
+
+        if node.side is waveform.Side.Right:
```

Comment:
can you remind me what's `Side.Right`?

Reply:
I wanted to reuse share the enums so that `Record` and `AllignedWaveform` could work together when generating padding.